src/commands.ts

```diff
@@ -52,17 +58,70 @@ function bzlMakeQueryQuickPickItem(queryItem: BazelQueryItem): BazelQueryQuickPi
 }
 
 async function bzlQuickPickQuery(query: string = '...', options?: QuickPickOptions) {
+
     return Window.showQuickPick(bzlQuery(query).then(deps => {
         return deps.map(bzlMakeQueryQuickPickItem);
-    }, err => {
-        Window.showQuickPick([], {
+    }).catch(async err => {
+        await Window.showQuickPick([], {
             placeHolder: "<ERROR>"
-        })
-        Window.showErrorMessage(err.toString())
+        });
         return [];
     }), options);
 }
 
+function bzlClearQueryDiagnostics() {
+    bazelDiagnosticsCollection.clear();
+}
+
+function bzlQueryErrorDiagnostics(error: Error) {
+    let errorStr = error.toString();
+    let errors = errorStr.split("\n");
+
+    let errorStart = "ERROR:";
+    for(let error of errors) {
+        if(error.startsWith(errorStart)) {
+            error = error.substr(errorStart.length)
+
+            let slashIndex = error.indexOf("/")
+            let colonIndex = error.indexOf(":")
+            let drivePrefix = ''
+            if(colonIndex > -1 && colonIndex < slashIndex) {
+                drivePrefix = error.substr(0, colonIndex+1).trim()
+                colonIndex = error.indexOf(":", colonIndex+1)
+            }
+
+            let [lineStr, colStr] = error.substr(colonIndex+1).split(":")
+
+            // Bazel returns non-zero based
+            let line = parseInt(lineStr)-1
+            let col = parseInt(colStr)-1
+
+            let startPosition = new Position(line, col)
+            let endPosition = new Position(line, col)
+            let range = new Range(startPosition, endPosition)
+            
+            let errorMessage = error.substr(
+                colonIndex + lineStr.length + colStr.length + 3
```

Comment:
Where does this 3 come from?

Reply:
```
ERROR: E:/projects/bazelregistry/sdl2/BUILD:22:1: syntax error at '}': expected ]
```
It's the 2 colons between the line and the column plus the whitespace character before `syntax`. Honestly, it's a pretty hacky parse.

Reply:
Is there a way to improve it?

Reply:
May I propose a regex parse, below a sample:

```js
const regex = /ERROR: (\w:)?([^:]*):(\d+):(\d+): (.*)/;
const errors = [
    "ERROR: E:/projects/bazelregistry/sdl2/BUILD:22:1: syntax error at '}': expected",
    "ERROR: /projects/bazelregistry/sdl2/BUILD:22:1: syntax error at '}': expected" 
];

let match;
for (const error of errors) {
    if ((m = regex.exec(error)) !== null) {
        let drivePrefix = m[1];
        let path        = m[2];
        let line        = m[3];
        let column      = m[4];
        let message     = m[5];

        console.log(JSON.stringify({
            drivePrefix: drivePrefix,
            path: path,
            line: line,
            column: column,
            message: message
        }, null, 2));
    }
}
```

and the result:
```json
{
  "drivePrefix": "E:",
  "path": "/projects/bazelregistry/sdl2/BUILD",
  "line": "22",
  "column": "1",
  "message": "syntax error at '}': expected"
}
{
  "path": "/projects/bazelregistry/sdl2/BUILD",
  "line": "22",
  "column": "1",
  "message": "syntax error at '}': expected"
}
```

Reply:
@zaucy Maybe you can integrate the solution proposed by @quentingodeau ?